src/hooks/useOverlayManager.tsx

```diff
@@ -1,39 +1,71 @@
-import React, { useCallback, useContext, useState } from 'react'
+import React, { useCallback, useContext, useState, useRef } from 'react'
 import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock'
+import styled from '@emotion/styled'
 import { css, Global } from '@emotion/react'
+import { zIndex } from '@/shared/theme'
 
 type OverlayManagerContextValue = {
-  overlayOpen: boolean
-  setOverlayOpen: (value: boolean, scrollbarGap?: number) => void
+  scrollLocked: boolean
+  setScrollLocked: (value: boolean, scrollbarGap?: number) => void
+  overlayContainerOpened: boolean
+  setOverlayContainerOpened: (value: boolean) => void
+  overlayContainerRef: React.RefObject<HTMLDivElement>
 }
+
+type OverlayContainerProps = {
+  isOpened?: boolean
+}
+
 const OverlayManagerContext = React.createContext<OverlayManagerContextValue | undefined>(undefined)
 OverlayManagerContext.displayName = 'OverlayManagerContext'
 
 export const OverlayManagerProvider: React.FC = ({ children }) => {
-  const [overlayOpen, setOverlayOpen] = useState(false)
+  const [scrollLocked, setScrollLocked] = useState(false)
+  const [overlayContainerOpened, setOverlayContainerOpened] = useState(false)
   const [scrollbarGap, setScrollbarGap] = useState(0)
+  const overlayContainerRef = useRef<HTMLDivElement>(null)
 
-  const handleOverlayOpen = (value: boolean, scrollbarGap?: number) => {
-    if (value === overlayOpen) {
+  const handleScrollLocked = (value: boolean, scrollbarGap?: number) => {
```

Comment:
I just noticed this, but there's actually a bug here (introduced by me):

`lockScroll` (old `handleOverlayOpen`) in the hook is created using `useCallback`. That's because it's a dependency to `useEffect` in `SearchOverlayView`, otherwise that `useEffect` would keep getting updated. *But*, those `useCallback`s actually have dependencies of their own (for example this function) which actually aren't memoized with `useCallback` so the `useEffect` will potentially get reexecuted when there's no need to. We need to put this function and `handleContainerOpened` into `useCallback` as well